torch/distributed/distributed_c10d.py

```diff
@@ -655,7 +655,6 @@ def _new_process_group_helper(
 
     if not group_name:
         group_name = str(_group_count)
-        _group_count += 1
```

Comment:
What is the reason to remove the group_count increment from here and move it below?

Reply:
Here the `_group_count` is updated only if the group does not have a name. The change is made to use `_group_count` to represent the total number of groups created. This has two cases. The first is when a non-participating rank tries to create a group. In this case, the `_group_count` is immediately incremented and the `GroupMember.NON_GROUP_MEMBER` is returned.

The second is when a legitimate process group is created. In this case, the `_group_count`` is incremented after the group is created at the end.